src/sql/workbench/contrib/commandLine/electron-browser/commandLine.ts

```diff
@@ -324,9 +368,26 @@ export class CommandLineWorkbenchContribution implements IWorkbenchContribution,
 		profile.setOptionValue('applicationName', applicationName);
 		profile.setOptionValue('databaseDisplayName', profile.databaseName);
 		profile.setOptionValue('groupId', profile.groupId);
+		// Set all advanced options
+		let advancedOptions = this.getAdvancedOptions(args.options, profile.getOptionKeyIdNames());
+		advancedOptions.forEach((v, k) => {
+			profile.setOptionValue(k, v);
+		});
 		return this._connectionManagementService ? this.tryMatchSavedProfile(profile) : profile;
 	}
 
+	private getAdvancedOptions(options: string, idNames: string[]): Map<string, string> {
+		let advancedOptionsMap = new Map<string, string>();
+		if (options) {
+			JSON.parse(options, (k, v) => {
+				if (!(k in idNames)) {
```

Comment:
Is there a reason you're not using the value if it exists already? Usually for stuff like this you would overwrite the original settings if the user specifies their own directly.

Reply:
It's not to check if it exists, but to discourage providing any key property names in advanced options (server/database/user/authtype etc.) that user has other properties to provide. To make it's purpose precisely to read advanced options only.

That reminds me I should also make sure password & azure access token are also ignored here. On it.